nixos/modules/services/development/livebook.nix

```diff
@@ -0,0 +1,75 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.livebook;
+in
+{
+  options.services.livebook = {
+    enableUserService = mkEnableOption "a user service for Livebook";
+
+    environmentFile = mkOption {
+      type = types.path;
+      description = lib.mdDoc ''
+        Environment file as defined in {manpage}`systemd.exec(5)` passed to the service.
+
+        This must contain at least `LIVEBOOK_PASSWORD`.  See `livebook server --help` for
+        other options.
+      '';
+    };
+
+    erlang_node_short_name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "livebook";
+      description = "A short name for the distributed node.";
+    };
+
+    erlang_node_name = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      example = "livebook@127.0.0.1";
+      description = "The name for the app distributed node.";
+    };
+
+    port = mkOption {
+      type = with types; nullOr port;
+      default = 8080;
+      description = "The port to start the web application on.";
+    };
+
+    cookie = mkOption {
```

Comment:
unless I remember incorrectly, this is supposed to be a secret. Having it in the nix store is not ideal.
if you don't mind, let's reduce the number of arguments in this module to the minimum (only address, and port) and just add another attribute called extraArguments or extraOptions, that contains an attribute set that can be merged with the others in gnucommandlineshell.
having cookie here would suggest that storing a secret here is a good idea.

Reply:
Perhaps an alternative would be to have a cookieFile option. Then users would be able to use something like [agenix](https://github.com/ryantm/agenix) to manage the cookie variable.

Reply:
And then the description for this option should probably include a big fat "DO NOT USE". Kind of like this warning gitea: https://github.com/NixOS/nixpkgs/blob/nixos-23.05/nixos/modules/services/misc/gitea.nix#L132-L134

Reply:
The Erlang magic cookie isn't meant to be a security feature, so it doesn't need to be secret.  Per [the manual](https://www.erlang.org/doc/reference_manual/distributed.html#security):

> "Security" here does not mean cryptographically secure, but rather security against accidental misuse, such as preventing a node from connecting to a cluster with which it is not intended to communicate.

That said, fewer explicit options sounds like less maintenance work and more future-proofing, so I'll do this.

Reply:
Done.